### PR TITLE
(BOLT-157) show docs for tasks

### PR DESCRIPTION
**Pull request by lucywyman**
This depends on merging #198 first.

Built gem with these changes is installed to hf3a1dxx7lajzf0.delivery.puppetlabs.net, if you'd like to ssh in and verify it works (at least until EOD today, 12/19/2017).

### CODE REVIEW

File: lib/bolt/outputter/human.rb [hunk 1]
```diff
@@ -93,6 +93,24 @@ def print_table(results)
...
+        task['parameters'].each do |k, v|
+          pretty_params << "- #{k}: #{v['type']}\n\t#{v['description']}\n"
+        end
+        task_info = <<-INFO
+#{task['name']} - #{task['description']}
+
+USAGE:
+#{task['usage']}
+
+PARAMETERS:
```
Comment (MikaelSmith) [inline lib/bolt/outputter/human.rb]: Should we avoid PARAMETERS if none are defined?

File: lib/bolt/outputter/human.rb [hunk 2]
```diff
@@ -93,6 +93,21 @@ def print_table(results)
         )
       end
 
+      # @param [Hash] A hash representing the task
+      def print_task(task)
+        pretty_params = ""
+        task_info = ""
+        task['parameters'].each do |k, v|
+          pretty_params << "- #{k}: #{v['type']}\n\t#{v['description']}\n"
```
Comment (lucywyman) [inline lib/bolt/outputter/human.rb]: This feels gross but also works and may be the best way to build the string?

Comment (MikaelSmith) [reply lib/bolt/outputter/human.rb]: It seems fine. You could use heredoc, but I think that'd be less clear if we actually want tabs.

File: spec/bolt/outputter/human_spec.rb [hunk 3]
```diff
@@ -40,6 +40,32 @@
...
+    outputter.print_task(task)
+    expect(output.string).to eq(<<-TASK_OUTPUT)
+cinnamon roll - A delicious sweet bun
+
+USAGE:
+Nourishment and happiness
+
+PARAMETERS:
+- icing: Cream cheese
+\tRich, tangy, sweet
```
Comment (lucywyman) [inline spec/bolt/outputter/human_spec.rb]: Also gross. `\t` just seemed like the best way to make sure editors don't accidentally break this test (as mine is prone to do with tabs). But definitely tell me if it's unacceptable.

File: lib/bolt/cli.rb [hunk 4]
```diff
@@ -515,6 +523,37 @@ def list_tasks
       raise Bolt::CLIError, "Failure while reading task metadata"
     end
 
+    def get_task_info(task_name)
+      task_info = {}
+      # TODO: should this formatting be done somewhere else? I think it makes
```
Comment (MikaelSmith) [inline lib/bolt/cli.rb]: Seems ok.

Comment (adreyer) [reply lib/bolt/cli.rb]: Long term maybe in puppet strings. I think it makes sense to generate the usage line in the human outputter and not display it with json output for now.

File: lib/bolt/outputter/human.rb [hunk 5]
```diff
@@ -93,6 +93,21 @@ def print_table(results)
         )
       end
 
+      # @param [Hash] A hash representing the task
+      def print_task(task)
```
Comment (adreyer) [inline lib/bolt/outputter/human.rb]: I think its less than ideal that `print_task` prints the metadata while `print_plan` prints the results of a plan.
it should either be` print_task_info` or `print_plan_results` probably

File: lib/bolt/outputter/human.rb (hunk 2 above)
Comment (adreyer) [reply lib/bolt/outputter/human.rb]: It will print an extra \n\t when there is no description which is probably wrong.

File: lib/bolt/cli.rb [hunk 6]
```diff
@@ -390,7 +391,14 @@ def execute(options)
       end
 
       if options[:mode] == 'task' && options[:action] == 'show'
-        outputter.print_table(list_tasks)
+        if options[:object]
+          outputter.print_task(get_task_info(options[:object]))
+        else
+          outputter.print_table(list_tasks)
+          outputter.print_message("Use `bolt task show <task-name>` to view
```
Comment (adreyer) [inline lib/bolt/cli.rb]: this `bolt task show <task>` elsewhere it should probably consistent

File: spec/bolt/outputter/human_spec.rb (hunk 3 above)
Comment (adreyer) [reply spec/bolt/outputter/human_spec.rb]: why not just use spaces?

File: lib/bolt/cli.rb (hunk 6 above)
Comment (lucywyman) [reply lib/bolt/cli.rb]: Not sure what you mean? Where else should I put the string?

Comment (MikaelSmith) [reply lib/bolt/cli.rb]: Updated the comment for clarity.

File: lib/bolt/cli.rb (hunk 4 above)
Comment (lucywyman) [reply lib/bolt/cli.rb]: Should the json output just display hash returned from `compiler.task_signature(task_name).task_hash` ? Or should I edit that at all?

File: spec/bolt/cli_spec.rb [hunk 7]
```diff
@@ -582,6 +582,45 @@ def stub_config(file_content = nil, base = nil)
...
+          options = {
+            mode: 'task',
+            action: 'show',
+            object: task_name
+          }
+          cli.execute(options)
+          expect(JSON.parse(@output.string)).to eq(
+            "name" => "sample::params",
+            "description" => "Task with parameters",
+            "executable" => "#{Dir.pwd}/spec/fixtures/modules/sample/tasks/params.sh",
```
Comment (lucywyman) [inline spec/bolt/cli_spec.rb]: This seems fragile. Is it better to remove the executable kv pair from the object here, remove it when we get the task info from the compiler, or to just leave it as is?

Comment (adreyer) [reply spec/bolt/cli_spec.rb]: Lets just not test that part of the output. I don't know how important showing the executable path is.

**Pull request merged**